examples/vision/mixUp.py

```diff
@@ -0,0 +1,219 @@
+
+    # Perform mixup on both images and labels
+    images = images_one * x_l + images_two * (1 - x_l)
+    labels = labels_one * y_l + labels_two * (1 - y_l)
+    return (images, labels)
+
+
+"""
+**Note** that here removing the vectorization part, we are combining two images to create
+a single one. Theoretically, we can combine as many we want but that comes at an
+increased computation cost. In certain cases, it may not help improve the performance as
+well.
+"""
+
+"""
+## Visualize new dataset
+"""
+
+# First create the new dataset using our `mix_up` utility
+train_ds_mu = train_ds.map(
+    lambda ds_one, ds_two: mix_up(ds_one, ds_two, alpha=0.2), num_parallel_calls=AUTO
+)
+
+# Now, we are ready to visualize
+sample_images, sample_labels = next(iter(train_ds_mu))
+plt.figure(figsize=(10, 10))
+for i, (image, label) in enumerate(zip(sample_images[:9], sample_labels[:9])):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(image.numpy().squeeze())
+    print(label.numpy().tolist())
+    plt.axis("off")
+
+"""
+## Model building
+"""
+
+
+def get_training_model():
+    model = tf.keras.Sequential()
+    model.add(L.Conv2D(16, (5, 5), activation="relu", input_shape=(28, 28, 1)))
+    model.add(L.MaxPooling2D(pool_size=(2, 2)))
+    model.add(L.Conv2D(32, (5, 5), activation="relu"))
+    model.add(L.MaxPooling2D(pool_size=(2, 2)))
+    model.add(L.Dropout(0.2))
+    model.add(L.GlobalAvgPool2D())
+    model.add(L.Dense(128, activation="relu"))
+    model.add(L.Dense(10, activation="softmax"))
+    return model
+
+
+"""
+For the sake of reproducibility, we serialize the initial random weights of our shallow
+network.
+"""
+
+initial_model = get_training_model()
+initial_model.save_weights("initial_weights.h5")
+
+"""
+## #1 Train model with a mixed up dataset
+"""
+
+model = get_training_model()
+model.load_weights("initial_weights.h5")
+model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
+model.fit(train_ds_mu, validation_data=test_ds, epochs=EPOCHS)
+_, test_acc = model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_acc * 100))
+
+"""
+## #2 Train model *without* a mixed up dataset
+"""
+
+model = get_training_model()
+model.load_weights("initial_weights.h5")
+model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
+# Notice that we are NOT using the mixed up dataset here
+model.fit(train_ds_one, validation_data=test_ds, epochs=EPOCHS)
+_, test_acc = model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_acc * 100))
+
+"""
+Readers are encouraged to try out mixup on different datasets from different domains and
+experiment with the $\lambda$ parameter. You are strongly advised to check out the
+original paper as well. In it, the authors present several ablation studies on mixup
+showing how it can improve generalization. Along with that, they also present results
+when you combine more than two images to create a single one.
```

Comment:
Suggestion: minor refactoring + add a link to the paper again

```suggestion
You are encouraged to try the mixup routine on different datasets from different domains and
experiment with various $\lambda$ parameters. You are strongly advised to check out the
[original paper](https://arxiv.org/abs/1710.09412) as well—the authors present several ablation
studies on mixup showing how it can improve generalization, as well as show their results of
combining more than two images to create a single one.
```